### ISSUE TITLE
camera.js unable to continue stream after taking picture

### ISSUE THREAD

**Issue by nickdibari** [nickdibari/PartyCanon#1]
During the execution of the stream cycle, the program will take a picture but then halt and sometimes crash. Why this happens and how to prevent it are currently unknown. What's curious about this is that it only raised an Error once, in all other cases it seemed to get hung up in the stream.

Traceback from the one actual error:
```
Unhandled rejection TypeError: Cannot read property 'payload' of undefined
    at Object.GetLiveviewDataJson [as GetLiveviewData] (/home/nick/Downloads/codecanon-tome-mm100-ws/howielib/howie/cam/MMJsonResponse.js:9:28)
    at MMJsonResponse (/home/nick/Downloads/codecanon-tome-mm100-ws/howielib/howie/cam/MMJsonResponse.js:96:29)
    at startRemoteShoot.then.then (/home/nick/Downloads/codecanon-tome-mm100-ws/howielib/howie/cam/MMCamera.js:464:12)
    at tryCatcher (/home/nick/Downloads/codecanon-tome-mm100-ws/howielib/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/home/nick/Downloads/codecanon-tome-mm100-ws/howielib/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (/home/nick/Downloads/codecanon-tome-mm100-ws/howielib/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromise0 (/home/nick/Downloads/codecanon-tome-mm100-ws/howielib/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (/home/nick/Downloads/codecanon-tome-mm100-ws/howielib/node_modules/bluebird/js/release/promise.js:693:18)
    at Async._drainQueue (/home/nick/Downloads/codecanon-tome-mm100-ws/howielib/node_modules/bluebird/js/release/async.js:133:16)
    at Async._drainQueues (/home/nick/Downloads/codecanon-tome-mm100-ws/howielib/node_modules/bluebird/js/release/async.js:143:10)
    at Immediate.Async.drainQueues (/home/nick/Downloads/codecanon-tome-mm100-ws/howielib/node_modules/bluebird/js/release/async.js:17:14)                          
    at runCallback (timers.js:672:20)                                             
    at tryOnImmediate (timers.js:645:5)                                           
    at processImmediate [as _immediateCallback] (timers.js:617:5) 
```

**Comment by nickdibari** [comment]
After talking to Mike, he suggested we implement a timeout on the snap command. In his words, he hypothesizes that "the issue is that the stopStream command returns immediately, but the stream isn’t actually stopped for an indeterminate amount of time (when the following frame actually is received.) So a delay could account for that but will depend on when the next frame is received."

Documentation on setting timeouts in Javascript: https://www.w3schools.com/jsref/met_win_settimeout.asp

**Comment by nickdibari** [comment]
After talking to Leslie, we came up with two ideas to solve this problem

1) Call functions to stop stream before picture is taken, and then resume stream upon successful retrieval of snapped image
    - Tried this before and did not seem to work

2) Implement alternative stream method that returns a Promise that resolves to a frame object
    - Unsuccessful, program was unable to run do to a Heap overflow. Wonder if executing statement in a loop causes memory issues

**Comment by nickdibari** [comment]
After code review with another hacker, discovered that we could just pull the image directly from the frame instead of snapping a picture. This seems to be the cleanest way to solve the problem and improves performance. Will need to keep an eye out to make sure the images are of good quality.

Fixed with f3876cfca466915e23a9243d1ee44d01eb537eaf
